core/src/main/java/org/apache/calcite/rex/RexUtil.java

```diff
@@ -2418,6 +2425,10 @@ protected RexNode lookup(RexNode expr) {
       return register(dynamicParam);
     }
 
+    @Override public RexNode visitNamedParam(RexNamedParam namedParam) {
```

Comment:
This seems right based on the other implementations. I don't really understand what it's for.

Reply:
I think it may also be used so that we can cache information like types. I'm not 100% sure but I saw something like that when diving into the namespace. I think this is correct regardless.